src/AutoPilotPlugins/APM/APMSensorsComponentSummary.qml

```diff
@@ -26,25 +27,88 @@ Item {
     Column {
         anchors.fill:       parent
 
+        VehicleSummaryRow {
+        labelText:  qsTr("Compasses:")
+        valueText: ""
+        }
+
         Repeater {
             model: 3
+            RowLayout {
+                Layout.fillWidth: true
+                width: parent.width
 
-            VehicleSummaryRow {
-                labelText:  qsTr("Compass ") + (index + 1) + ""
-                valueText:  sensorParams.rgCompassAvailable[index] ?
+                QGCLabel {
+
+                    text:  sensorParams.rgCompassAvailable[index] ?
                                 (sensorParams.rgCompassCalibrated[index] ?
-                                     (sensorParams.rgCompassPrimary[index] ? "Primary" : "Secondary") +
+                                     getPriority(index) +
                                      (sensorParams.rgCompassExternalParamAvailable[index] ?
                                           (sensorParams.rgCompassExternal[index] ? ", External" : ", Internal" ) :
                                           "") :
                                      qsTr("Setup required")) :
                                 qsTr("Not installed")
+
+                    function getPriority (index) {
+                        if (sensorParams.rgCompassId[index].value == sensorParams.rgCompassPrio[0].value) {
+                            return "Primary"
+                        }
+                        if (sensorParams.rgCompassId[index].value == sensorParams.rgCompassPrio[1].value) {
+                            return "Secondary"
+                        }
+                        if (sensorParams.rgCompassId[index].value == sensorParams.rgCompassPrio[2].value) {
+                            return "Tertiary"
+                        }
+                        return "Unused"
+                    }
+                }
+
+                APMSensorIdDecoder {
+                    horizontalAlignment:    Text.AlignRight
+                    Layout.alignment:       Qt.AlignRight
+
+                    fact: sensorParams.rgCompassPrio[index]
+                }
+
+                function getPriority (index) {
```

Comment:
Why there is a duplicated function here ?

Reply:
bad rebase :sweat_smile:

Reply:
This is still here

Reply:
what about now?